packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx

```diff
@@ -82,14 +82,19 @@ const AccountRecoveryConfirmKey = ({
     if (errorMessage) {
       setErrorMessage('');
     }
-    // strip out any characters other than letters, numbers, spaces
-    // and only allow single spaces
+
+    // allow only alphanumeric characters and single spaces
     const filteredCode = e.target.value
       .replace(/[^a-zA-Z0-9\s]/g, '')
       .replace(/\s{2,}/g, ' ');
-    // only count letters and numbers to assess if submit should be disabled
-    setIsSubmitDisabled(removeSpaces(filteredCode).length !== 32);
-    e.target.value = filteredCode;
+    // limit the length to 32 characters (excluding spaces)
+    const truncatedCode = filteredCode.slice(
+      0,
+      32 + (filteredCode.length - removeSpaces(filteredCode).length)
```

Comment:
Just a small nit, I think setting `removeSpaces(filteredCode).length` to `const codeWithoutSpaces` and setting a `const RECOVERY_KEY_LENGTH = 32` at the top of this file (or whatever you want to name these) could help with clarity.

Reply:
I reviewed this and extracted it into a function. While we re-testing I noticed an error that would sometimes allow 33 alphanumeric characters to be typed in.